packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -160,19 +155,18 @@ class TransactionService {
     if (type === PAGE_LOAD) {
       tr.isHardNavigation = true
 
+      if (tr.name === NAME_UNKNOWN) {
+        const pageUrl = stripQueryStringFromUrl(window.location.href)
+        if (pageUrl) {
+          tr.name = pageUrl
+        }
+      }
       if (perfOptions.pageLoadTraceId) {
         tr.traceId = perfOptions.pageLoadTraceId
       }
       if (perfOptions.pageLoadSampled) {
         tr.sampled = perfOptions.pageLoadSampled
       }
-      /**
-       * Retriving the name before transaction ends should reflect
-       * the correctg page load transaction name
-       */
-      if (tr.name === NAME_UNKNOWN && perfOptions.pageLoadTransactionName) {
```

Comment:
This is totally unnecessary, We send the `pageLoadTransactionName` from `apm-base` which already handles this  scenario.